adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -876,6 +876,17 @@ func (apiReconciler *APIReconciler) getAPIsForScope(obj k8client.Object) []recon
 func (apiReconciler *APIReconciler) getAPIsForBackend(obj k8client.Object) []reconcile.Request {
 	ctx := context.Background()
 	backend, ok := obj.(*dpv1alpha1.Backend)
+	retryConfig := backend.Spec.Retry
+	if retryConfig != nil {
+		if int32(retryConfig.Count) < 0 {
+			loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(3117, utils.NamespacedName(backend).String()))
+		}
+		for _, statusCode := range retryConfig.StatusCodes {
+			if statusCode > 598 || statusCode < 401 {
+				loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(3116, utils.NamespacedName(backend).String()))
+			}
+		}
```

Comment:
please move this to validation hook

Reply:
This will be done once this PR https://github.com/wso2/apk/pull/1313 got merged as it contains the validation files already